Kernel/include/semaphores.h

```diff
@@ -0,0 +1,78 @@
+#ifndef _SEMAPHORE_H_
+#define _SEMAPHORE_H_
+
+#include <stddef.h>
+#include <lib.h>
+#include <memoryManager.h>
+#include <resourceNamerADT.h>
+#include <waitQueueADT.h>
+#include <kernelTypes.h>
+#include <scheduler.h>
+
+#define MAX_SEMAPHORES 128
```

Comment:
Careful, remember you're not using index 0 and an `int8_t`'s max value is 127.

Reply:
You are right